service/lib/agama/autoyast/converter.rb

```diff
@@ -111,41 +116,19 @@ def tmp_profile_path
 
       # @return [Hash] Agama profile
       def export_profile(profile)
-        users = Agama::AutoYaST::UsersConverter.new(profile)
-        {
-          "software" => export_software(profile.fetch_as_hash("software")),
-          "storage"  => export_storage(profile.fetch_as_array("partitioning")),
-          "root"     => users.root,
-          "user"     => users.user
-        }
-      end
-
-      # @param drives [Array<Hash>] Array of drives in the AutoYaST partitioning section
-      def export_storage(drives)
-        # TODO: rely on AutoinstProfile classes
-        devices = drives.each_with_object([]) do |d, all|
-          next unless d["device"]
-
-          all << d["device"]
-        end
-        return {} if devices.empty?
-
-        { "bootDevice" => devices.first }
-      end
-
-      # @param profile [Hash] Software section from the AutoYaST profile
-      def export_software(profile)
-        product = profile.fetch_as_array("products").first
-        patterns = profile.fetch_as_array("patterns")
-        return {} unless product
-
-        { "product" => product, "patterns" => patterns }
-      end
-
-      # @param profile [Hash] Users section from the AutoYaST profile
-      def export_root(profile)
-        users = Agama::AutoYaST::UsersConverter.new(profile)
-        users.root
+        user = Agama::AutoYaST::UserReader.new(profile)
+        root = Agama::AutoYaST::RootReader.new(profile)
+        software = Agama::AutoYaST::SoftwareReader.new(profile)
+        product = Agama::AutoYaST::ProductReader.new(profile)
+        l10n = Agama::AutoYaST::L10nReader.new(profile)
+        storage = Agama::AutoYaST::StorageReader.new(profile)
+
+        user.read
+          .merge(root.read)
+          .merge(software.read)
+          .merge(product.read)
+          .merge(storage.read)
+          .merge(l10n.read)
```

Comment:
NP, just matter of taste, but I would write it in more easier way for extend like:
```
READERS = ["user", "root", "software", ... ] # extend list to add support for more sections
def export_profile(profile)
  sections.each_with_object({}) do |section, result|
    require "agama/autoyast/#{section}_reader}"
    klass = "#{section}_reader".split("_").map(&:capitalize).join
    reader = Agama::AutoYaST.get_const(klass).new(profile)
    result.merge(reader.read)
   end
end
```

this way it is really just single line to extend it with new section as opposite to current code which require change in three places.

Reply:
btw I can even reduce it to no line in converter if we have something like `agama/autoyast` file that contain all classes and then filter it by known readers.

Reply:
Good point! I decided to go with the `READERS` variable approach because it is more explicit.